.buildkite/scripts/benchmark/main.sh

```diff
@@ -107,6 +107,9 @@ pull_images() {
     docker pull "docker.elastic.co/logstash/logstash:$LS_VERSION"
   else
     LS_VERSION=$( curl --retry-all-errors --retry 5 --retry-delay 1 -s https://artifacts-api.elastic.co/v1/versions | jq -r ".versions[-1]" )
+    if [[ ! $LS_VERSION == *-SNAPSHOT ]]; then
+      LS_VERSION="${LS_VERSION}-SNAPSHOT"
+    fi
```

Comment:
If we cannot guarantee the position of the item we are looking for in the list coming back from the artifacts API, I would rather tune the query to select the one we want instead of blindly appending `SNAPSHOT` to an artifact and _hoping_ it matches one of the artifacts in the list.

```suggestion
    # select the SNAPSHOT artifact with the highest semantic version number
    LS_VERSION=$( curl --retry-all-errors --retry 5 --retry-delay 1 -s https://artifacts-api.elastic.co/v1/versions | jq -r '.versions | map(select(endswith("-SNAPSHOT"))) | max_by(rtrimstr("-SNAPSHOT")|split(".")|map(tonumber))' )
```